daemons/gptp/common/ieee1588port.cpp

```diff
@@ -598,7 +598,8 @@ void IEEE1588Port::processEvent(Event e)
 
 	case STATE_CHANGE_EVENT:
 		if (!automotive_profile) {       // BMCA is not active with Automotive Profile
-			if ( clock->getPriority1() != 255 ) {
+			if ( clock->getPriority1() != 255 && qualified_announce != NULL ) {
```

Comment:
I think checking qualified_announce here is redundant;  it's set right before STATE_CHANGE_EVENT is queued for this port and it's also checked below when `calculateERBest()` is called on each port in your addition below.  I can't see that it's accessed between here and the check below.